[docs] Update release statuses (#48545)
expo/expo: docs/pages/more/release-statuses.mdx

---
title: Release statuses
description: Learn about the experimental, alpha, preview, beta, stable, and deprecated release statuses used across the Expo SDK, Expo Router, EAS, and Expo CLI.
---

Expo uses various release statuses to indicate the stability and readiness of its tools and services. Understanding these statuses can help you make informed decisions about which features and versions to use in your projects.

These statuses are not a single pipeline that every feature moves through. Each product area uses the statuses that fit how it ships:

| Product area                         | Statuses you'll mostly see         |
| ------------------------------------ | ---------------------------------- |
| Expo SDK libraries and Expo Router   | [Alpha](#alpha), [Beta](#beta)     |
| EAS services                         | [Preview](#preview), [Beta](#beta) |
| Expo CLI and other development tools | [Experimental](#experimental)      |

> **info** **Note:** Any feature can become [deprecated](#deprecated) at the end of its life, and individual APIs inside otherwise stable libraries can be [experimental](#experimental).

> ## Experimental

Reviewer: side-note: We should differentiate this from `experiments` in the `app.json` maybe? I'm afraid people may get the two mixed up, since we didn't name this `featureFlags` (or something similar)

**What to expect:**

- May change significantly or be removed entirely in any release
- APIs are subject to change without notice or deprecation warnings

> - Implementation of experimental features should be treated as a proof of concept rather than final

Reviewer 2: Use imeparative voice
```suggestion
- Treat the implementation of an experimental feature as a proof of concept rather than final
```

- Can be used in production with thorough testing

The purpose of a preview is to share new functionality early and gather feedback about a specific slice before building it out further. Your input directly helps shape the direction of the feature. [Share your feedback](https://expo.dev/contact) to help us prioritize what to build next.

## Beta

Beta features are feature-complete and undergoing final validation before a stable release. You'll mostly see this status on EAS services and on SDK libraries that are close to stable.

**What to expect:**

- Core functionality is complete
- Breaking changes are possible but unlikely unless critical issues are found
- Can be used in production with thorough testing

Beta features are ready for real-world testing. While we don't expect major changes, we recommend testing thoroughly if you plan to use them in production. [Your feedback](https://expo.dev/contact) during this stage helps us catch any remaining issues before the stable release.

## Stable

Features without a status badge are stable and fully released for production use.

**What to expect:**

- Production-ready with full support

> - As per semantic versioning, breaking changes of stable features will only occur in new major versions

Reviewer 2: Use present tense

```suggestion
- Per semantic versioning, breaking changes to stable features only occur in new major versions
```

## Deprecated

Deprecated features are no longer recommended for use and will be removed in a future release. We provide deprecation warnings to give developers time to transition away from these features.

**What to expect:**

- Documentation and code show deprecation warnings

> - We may not accept feedback or issues for deprecated APIs

Reviewer 2: Avoid first-person plural in a non-tutorial page

```suggestion
- Expo may not accept feedback or issues for deprecated APIs
```

Author: We do tend to use when "we" represents Expo team. there are plenty of pages that do that. Just an FYI.

> - Removal of deprecated features and APIs will happen in a new SDK release

Reviewer 2: Use present tense

```suggestion
- Removal of deprecated features and APIs happens in a new SDK release
```

**What to expect:**

- May change significantly or be removed entirely in any release

> - APIs and implementation can change without notice or deprecation warnings

Reviewer: ```suggestion
- Implementation of experimental features should be treated as a proof of concept rather than final
```

Not quite sure how to express this, but expands on the note below

> - May have known bugs, missing functionality, or performance issues

Reviewer: I wouldn't remove this, as we often point this out in issue reports. The main conflict we often run into with the CLI/Devtools is that experimental features are not reflective of final implementations, and should be treated as a proof of concept.

This means that if we discover bugs or issues, they may not necessarily be fixed

**What to expect:**

> - APIs can have breaking changes without a major version bump

Reviewer: I kind of disagree with "can have" over "are subject to" for the term "breaking changes" exactly. It doesn't sound quite right to me, and "have" also implies that the APIs may have changes themselves, when what we're expressing is that any given API may itself break, as in, be removed/changed/added/etc

```suggestion
- APIs are subject to breaking changes without new SDK release
```

nit on "major version bump". We could explicitly say "SDK release" to make this more approachable?

- Implementation may change substantially based on feedback
- Not recommended for production apps

Alpha features are opportunities to influence the future of Expo. We encourage you to test these features in development environments and [provide feedback](https://expo.dev/contact) to help us refine them before they reach a wider audience.

## Preview

Preview features provide an early look at a focused slice of new functionality. They are designed with minimal overhead and limited scope, but they are not yet feature-complete. You'll mostly see this status on new EAS services.

**What to expect:**

- A focused slice of functionality, not the full feature set

> - APIs may evolve as the feature is built out

Reviewer: nit: "evolve" is very imprecise. We should check whether we can make a more specific guarantee here. (e.g. are we avoiding breaking changes?)

- Can be used in production with thorough testing

## Beta

Beta features are feature-complete and undergoing final validation before a stable release. You'll mostly see this status on EAS services and on SDK libraries that are close to stable.

**What to expect:**

> - Core functionality is complete and the API shape is mostly settled

Reviewer: nit: as per the above, "mostly settled" is imprecise. Do we avoid breaking changes? I assume so. Do we allow for additions w/o a minor release or replacement APIs with no deprecation notices? Maybe?

- Breaking changes are possible but unlikely unless critical issues are found
- Can be used in production with thorough testing

## Stable

Features without a status badge are stable and fully released for production use.

**What to expect:**

- Production-ready with full support

> - SDK libraries follow semantic versioning, so breaking changes only occur in new major versions

Reviewer: ```suggestion
- As per semantic versioning, breaking changes of stable features will only occur in new major versions
```

## Deprecated

Deprecated features are no longer recommended for use and will be removed in a future release. We provide deprecation warnings to give developers time to transition away from these features.

**What to expect:**

- Documentation and code show deprecation warnings

> - No new features or improvements will be made

Reviewer: ```suggestion
- We may not accept feedback or issues for deprecated APIs
```

I'd prefer we just state what this means for users concretely

> - For SDK libraries, removal happens in a new major version

Reviewer: ```suggestion
- Removal of deprecated features and API will happen in a new SDK release
```
